drivers/adc/adc_stm32.c

```diff
@@ -1628,6 +1747,13 @@ static const struct adc_driver_api api_stm32_driver_api = {
 	_CONCAT(ADC_STM32_CLOCK_PREFIX(x), ADC_STM32_DIV(x))
 #endif
 
+/* Macro to check if the ADC instance clock setup is correct */
+#define ADC_STM32_CHECK_DT_CLOCK(x)                                                                \
+	BUILD_ASSERT(                                                                              \
+		IS_EQ(ADC_STM32_CLOCK(x), SYNC) ||                                                 \
+			(IS_ENABLED(STM32_ADC_DOMAIN_CLOCK_SUPPORT) && DT_INST_NUM_CLOCKS(x) > 1), \
```

Comment:
I think this could be simplified with `BUILD_ASSERT(IS_EQ(ADC_STM32_CLOCK(x), SYNC) || (DT_INST_NUM_CLOCKS(x) > 1)),`
I don't think `IS_ENABLED(STM32_ADC_DOMAIN_CLOCK_SUPPORT)` brings more information than `(DT_INST_NUM_CLOCKS(x) > 1)`

Reply:
@gautierg-st Isn't that define true if any of the instances has a second clock defined ? DT_INST_NUM_CLOCKS(x) is the same as config->pclk_len for a specific instance.

Reply:
@gautierg-st Updated.